LibreNMS/Util/Rewrite.php

```diff
@@ -155,9 +155,13 @@ public static function readableMac($mac)
      */
     public static function readableOUI($mac)
     {
-        $key = 'OUIDB-' . (substr($mac, 0, 6));
+        $cached = Cache::get('OUIDB-' . (substr($mac, 0, 6)), '');
+        if ($cached == 'IEEE Registration Authority') {
+            // Then we may have a shorter prefix, so let's try them one ater the other, ordered by probability
+            return Cache::get('OUIDB-' . (substr($mac, 0, 9)), Cache::get('OUIDB-' . (substr($mac, 0, 7))));
```

Comment:
```php
return Cache::get('OUIDB-' . substr($mac, 0, 9)) ?: Cache::get('OUIDB-' . substr($mac, 0, 7));
```

This will prevent cache::get from being called twice 100% of the time.

Reply:
right ! I don't know why it took me so long to get it :D